packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.tsx

```diff
@@ -134,11 +135,23 @@ export class DatePicker extends BaseComponent<IDatePickerProps, IDatePickerState
     let { formatDate, isRequired, strings, value } = nextProps;
     const errorMessage = (isRequired && !value) ? (strings.isRequiredErrorMessage || '*') : null;
 
-    this.setState({
-      selectedDate: value || new Date(),
-      formattedDate: (formatDate && value) ? formatDate(value) : '',
-      errorMessage: errorMessage
-    });
+    if (errorMessage != null) {
+      this.setState({
+        errorMessage: errorMessage
+      });
+    }
+    else {
+      // Check if the date value changed from old props value, i.e., if indeed a new date is being
+      // passed in and only then update selectedDate
+      let oldValue = this.props.value;
+      if (!compareDates(oldValue, value)) {
+        this.setState({
+          selectedDate: value || new Date(),
+          formattedDate: (formatDate && value) ? formatDate(value) : '',
```

Comment:
Good catch. The new code would not honor the new format I guess... We could change the code to allow that but  then this means the bug will still repro in this case. Note that we cannot just use this.state.selectedDate here because this value may still have the old value given that setState only queues state changes and the value at this point may not be committed yet. I looked for way to force a state flush in react but couldn't really find anythings. Thoughts on what else we could do here?